api/v1alpha1/stage_types.go

```diff
@@ -423,6 +423,10 @@ type KustomizeImageUpdate struct {
 	//
 	// +kubebuilder:validation:Optional
 	UseDigest bool `json:"useDigest" protobuf:"varint,3,opt,name=useDigest"`
+	// NewName specifies a container image name override when setting the image
+	//
+	// +kubebuilder:validation:Optional
+	NewName string `json:"newName" protobuf:"bytes,5,opt,name=newName"`
```

Comment:
I think omitempty should usually be set on optional fields. Weirdly, controller-gen prioritizes that over the kubebuilder markers. It would probably be ok here, but we should do this for consistency.

Reply:
Apart from this, LGTM.

Reply:
Nice catch. Agree